src/workers/telegram_bot.rs

```diff
@@ -0,0 +1,312 @@
+use crate::core::{Variables, WorkerMessage};
+use ngrok::config::TunnelBuilder;
+use ngrok::prelude::{TunnelExt, UrlTunnel};
+use reqwest::blocking::Client;
+use roboplc::controller::{Context, WResult, Worker};
+use roboplc::event_matches;
+use roboplc_derive::WorkerOpts;
+use std::fmt::Debug;
+use std::sync::Arc;
+use std::thread::sleep;
+use std::time::Duration;
+use teloxide::dispatching::{Dispatcher, HandlerExt, UpdateFilterExt};
+use teloxide::macros::BotCommands;
+use teloxide::prelude::{ChatId, Message, Request, Requester, ResponseResult, Update};
+use teloxide::types::InputFile;
+use teloxide::utils::command::BotCommands as UtilsBotCommands;
+use teloxide::{dptree, Bot};
+use tokio::net::ToSocketAddrs;
+use tokio::runtime::Runtime;
+use tokio::sync::oneshot;
+use tracing::{debug, error, info, warn};
+
+#[derive(WorkerOpts)]
+#[worker_opts(cpu = 2, priority = 80, scheduling = "fifo", blocking = true)]
+pub struct BotWorker {}
+
+impl Worker<WorkerMessage, Variables> for BotWorker {
+    fn run(&mut self, context: &Context<WorkerMessage, Variables>) -> WResult {
+        while !check_internet_connection() {
+            info!("No internet connection. Retrying in 5 seconds...");
+            sleep(Duration::from_secs(5));
+        }
+
+        info!("Internet connection established. Starting bot...");
+        let result = Runtime::new().unwrap().block_on(bot(context));
+        Ok(result)
+    }
+}
+
+fn check_internet_connection() -> bool {
+    info!("Checking internet connection...");
+    let client = Client::builder()
+        .timeout(Duration::from_secs(10))
```

Comment:
I assume 10 second to match simple internet connection check.

Reply:
I use 10 sec because in some cases after startup device necessary some time for connecting Raspberry Pi to Wi-Fi (5 sec not always enough for instance, 10 sec - better). In general - need to find other way to check internet connections or not start telegram worker until internet connect.
I added it to my TODO.